subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java

```diff
@@ -14,14 +14,15 @@
  * limitations under the License.
  */
 
-package org.gradle.api.internal.changedetection.state;
+package org.gradle.internal.fingerprint.impl;
```

Comment:
If we start using the `impl` pattern to hide implementation details for internal types, then I think we should make a proposal about it on `engineering@`.

Reply:
Sounds like a good idea.

What is your suggestion for this PR? Move everything to `org.gradle.internal.fingerprint`?

Reply:
I think it's fine to make the move to `impl` where it makes sense. We should just not forget to mention this to others. If there's disagreement, we can easily clear things up after the fact, too.